src/Chapter2.hs

```diff
@@ -336,7 +349,11 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList i j l
+    | i > length l || i < 0 || j < 0 = []
+    | i <= j && j - 1 > length l = (l !! i) : subList (i + 1) (length l - 1) l
+    | i <= j = (l !! i) : subList (i + 1) j l
+    | otherwise = []
```

Comment:
Interesting solution! ✔️ 

Unfortunately, it is not efficient. Using `length` and `!!` is expensive. The `length` function should walk through each element of the list in order to get the size. And `!!` also should go one-by-one to the element each time.

You can implement this function without recursion, by using the combination of [`take`](https://hackage.haskell.org/package/base/docs/Prelude.html#v:take) and [`drop`](https://hackage.haskell.org/package/base/docs/Prelude.html#v:drop) functions 🙂